### ISSUE TITLE
Feature Request - Tabbed layout

### ISSUE THREAD

**Issue by sfrique** [leftwm/leftwm#681]
Hello,

Is there any plan to add a tabbed layout?

https://github.com/xmonad/xmonad-contrib/blob/master/XMonad/Layout/Tabbed.hs
Right now I am using Monocle, but I can't see the list of open windows on that workspace tag.

I mainly use this for a workspace/tag with all my chats/emails   (telegram/whatsapp/slack and etc)  that way I can keep all my communications on a single tag/workspace.
And usually it's prettier to have the chats on a bigger window.

If anyone has a suggestion on how to improve that workflow I appreciate it as well.
I wish I could get all the apps in a third party without losing functionality. I've tried a bit some options, none were good enough, but I am open to suggestions!

I can try to implement it as well, but i have no experience with rust nor I am a great programmer =P

Thanks

**Comment by VuiMuich** [comment]
Welcome adventurer!

I am sorry to only being able to suggest some workaroundish solution.

To get a list of the windows you cam use `wmctrl -l` and process it in a shellscript or with awk to display it in a widget like eww, or in you bar.
Unfortunately the flags `wmctrl -a` and `wmctrl -R` currently break the internal window list and require a `SoftReload` to get leftwm working correct again. (Will check on that later)

Besides the layout manager currently receiving a major rewrite and therefore any new layouts should possibly wait until this got merged, I actually think this really is not feasible to implement as an actual layout. <Will edit in further reasoning here later>

**Comment by sfrique** [comment]
If leftwm can't generate the tab-bars, that's fine. 
I could add some sort of conky or another bar for the info I need

I will wait for the layout rewrites, after it I can start to think of adding more layouts, being easy to add layouts is great.

I also saw the config will be changing, so I will be waiting for these changes to actually move to leftwm and test everything, cause it takes quite a while to get my WM to work as I want, but I am not happy with my current `xmonad` config file, it's a mess hahah

Anyway, thanks!

Closing this!
